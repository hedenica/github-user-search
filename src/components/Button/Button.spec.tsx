import { render, screen } from '@testing-library/react';

import { Button } from '.';

describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button title="Click me" />);

    expect(container).toMatchSnapshot();
  });

  describe('When prop disabled is passed', () => {
    it('renders as disabled', () => {
      render(<Button title="Disabled button" isDisabled />);

      const disabledButton = screen.getByText(/disabled button/i);

      expect(disabledButton).toBeDisabled();
    });
  });
});
