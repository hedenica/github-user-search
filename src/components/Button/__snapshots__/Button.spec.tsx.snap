// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class=""
    type="button"
  >
    Click me
  </button>
</div>
`;
